### PR TITLE
Refactor/remove platform plugin interface, for #458

### LINKED ISSUE #458: Simplify/remove plugin interface (remove interface, use LocalApp)
## What happened (or feature request):

We've had a longstanding complexity in ddev that is related to past unmet aspirations in the drud tool. In pkg/plugins we have a fancy plugin/interface pattern used in [types.go](https://github.com/drud/ddev/blob/master/pkg/plugins/platform/types.go) that only makes the code more complex, and doesn't add value. 

Recommendation: Lose the App interface and just use LocalApp in its place, lose GetPluginApp(). This is a very straightforward change, will take just a few minutes or hours, plus verification. I think all tests will continue without any issues. We can do this now and it will be something that will make maintenance, updates, and training on ddev code easier for all our future.

### PR DESCRIPTION
**Pull request by rfay**
## The Problem/Issue/Bug:

#458 specifies refactoring ddev to remove the obsolete and arcane plugin interface for platforms. This PR does that.

## How this PR Solves The Problem:

* LocalApp is renamed to DdevApp and moved to the ddevapp package.
* platform package is removed
* The interface abstraction is removed.

## Manual Testing Instructions:

Build or download ddev and put it through a few tests. Everything should work.

## Automated Testing Overview:

All existing tests are left in place, no changes.

## Related Issue Link(s):

OP #458

### CODE REVIEW

**Comment by rfay** [comment]
IMO this is ready for review. I added @beeradb and @tannerjfco to reviewers of course because of their extensive history with it. But @cweagans should review and manually test because this is what it will look like going forward.

I hope we can get this in pretty fast if it's satisfactory because it's one of those things that will break everything else.

**Comment by rickmanelius** [comment]
@rfay I'm also assuming no impact on the user since all existing tests are still in place?

**Comment by rfay** [comment]
Yes, this is internal code reorganization and simplification only, and has absolutely no impact on the user.

File: pkg/ddevapp/ddevapp_test.go [hunk 1]
```diff
@@ -59,14 +58,19 @@ var (
...
-	err := dockerutil.EnsureNetwork(client, dockerutil.NetName)
-	if err != nil {
-		log.Fatal(err)
-	}
+	// Ensure the ddev directory is created before tests run.
+	_ = util.GetGlobalDdevDir()
+
+	// Since this test may be first time ddev has been used, we need the
+	// ddev_default network available. This would normally be done in a
+	// TestMain, so can be moved to one when we need one.
```
Comment (tannerjfco) [inline pkg/ddevapp/ddevapp_test.go]: this _is_ in a TestMain, so this comment is a bit odd

File: pkg/ddevapp/ddevapp_test.go [hunk 2]
```diff
@@ -59,14 +58,19 @@ var (
...
+	dockerutil.EnsureDdevNetwork()
 
-	count := len(platform.GetApps()["local"])
+	// Avoid having sudo try to add to /etc/hosts.
+	// This is normally done by Testsite.Prepare()
+	_ = os.Setenv("DRUD_NONINTERACTIVE", "true")
+
+	count := len(ddevapp.GetApps())
 	if count > 0 {
 		log.Fatalf("Local plugin tests require no sites running. You have %v site(s) running.", count)
```
Comment (tannerjfco) [inline pkg/ddevapp/ddevapp_test.go]: nit, err message could use an update

**Comment by rfay** [comment]
Thanks so much to everybody for looking at this. I took another pass through and fixed Tanner's nits, and also tried to update most occurrences of "local" that were just flat obsolete. I'll spin off the usage of DEPLOY_NAME for drush (and the failure of drush.settings.php) into another issue.

**Comment by cweagans** [comment]
🎉

**Pull request merged**